mailchimp/wordpress
# Enhancement/update critical flows with e2e test coverage

### Description of the Change

Update E2E Cypress tests to cover all critical testing flows.

- [Critical flows test plan](https://github.com/mailchimp/wordpress/wiki/Critical-flows-Mailchimp)
  - [Detailed notes for each item](https://github.com/mailchimp/wordpress/wiki/E2E-Checklist-of-critical-flow-for-Mailchimp-(can-be-deleted-later)#not-started--mailchimp-setting--list-options--use-javascript-support)

#### Enhancement/add mailchimp api client - #99
- Add a FE Mailchimp client to Cypress tests to verify Mailchimp account interactions. Ensures successful form submission and full integration testing.
- Use `.env.test` for env variables
- Add health check for tests if Mailchimp client is unable to make requests
- Modularize Cypress commands

#### ‼️ Environment changes needed ‼️
- Need to generate an API key for the test Mailchimp user in the CI/CD pipeline.
- Need to attach that API key to CYPRESS_MAILCHIMP_API_KEY as a Github secret
- Need to attach the server prefix to CYPRESS_MAILCHIMP_API_SERVER_PREFIX as a Github secret


Closes #61
Closes #86 

### How to test the Change

Follow the E2E testing instructions in the [readme](https://github.com/mailchimp/wordpress?tab=readme-ov-file#e2e-tests) to run the Cypress tests. In addition, these tests should run automatically as part of the projects CI/CD flow.

### Changelog Entry

> Added - New tests to cover E2E critical flows
> Changed - Existing tests to more robustly cover E2E critical flows. Modularized reusable testing code.

### Credits

Props @MaxwellGarceau 

### Checklist:
